core/network/endpoint_bindings.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package network
+
+import "github.com/juju/errors"
+
+// MapWithSpaceNames returns the current bindingMap with space names rather than ids.
+func MapBindingsWithSpaceNames(bindingsMap map[string]string, lookup SpaceInfos) (map[string]string, error) {
+	// Handle the fact that space name lookup can be nil or empty.
+	if lookup == nil || (len(bindingsMap) > 0 && len(lookup) == 0) {
```

Comment:
This can just be `if (len(bindingsMap) > 0 && len(lookup) == 0)`. The nil check is superfluous

Reply:
Technically not quite, but in practice you're right